packages/next/src/server/image-optimizer.ts

```diff
@@ -795,9 +795,12 @@ export async function fetchInternalImage(
   ) => Promise<void>
 ): Promise<ImageUpstream> {
   try {
+    // Coerce HEAD to GET to avoid issues with the image optimizer
+    const method = _req.method === 'HEAD' ? 'GET' : _req.method
+
     const mocked = createRequestResponseMocks({
       url: href,
-      method: _req.method || 'GET',
+      method: method || 'GET',
```

Comment:
```suggestion
    const method = (!_req.method || _req.method === 'HEAD') ? 'GET' : _req.method

    const mocked = createRequestResponseMocks({
      url: href,
      method,
```

Can do this in one line